phpBB/phpbb/storage/storage.php

```diff
@@ -18,43 +18,110 @@
  */
 class storage
 {
+	/**
+	 * @var \phpbb\storage\adapter\adapter_interface
+	 */
 	protected $adapter;
 
-	public function __construct($adapter)
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\storage\adapter_factory	$factory
+	 * @param string							$storage_name
+	 */
+	public function __construct(adapter_factory $factory, $storage_name)
 	{
-		$this->adapter = $adapter;
+		$this->adapter = $factory->get($storage_name);
```

Comment:
This is a very good idea, and solves a problem that i already knew i was going to have